release-notes.txt

```diff
@@ -1,3 +1,527 @@
++ drivers/periph/uart: add periph_uart_tx_ondemand feature (#20108)
+* cpu/avr8_common: allow to build with !periph_pm (#20247)
+* cpu/avr8_common: fix build with !core_thread (#20240)
+* cpu/nrf5x_common: clean up UART implementation (#20111)
+* cpu/sam0/uart: implement uart_pin_cts()/uart_pin_rts() (#20171)
+* cpu/sam0_common: implement periph_gpio_ll and periph_gpio_ll_irq (#20189)
+* cpu/stm32/periph_spi: only perform DMA transfer above threshold (#20214)
+* cpu/stm32: uart: don't do DMA for small transfers (#20230)
+
+Device Drivers
+--------------
+
++ driver/mtd: mtd_init add documentation for return value (#20033)
++ drivers/periph_timer: add timer_query_freqs() (#16349)
+* drivers/mtd_flashpage: improve _write_page (#20173)
+* drivers/mtd: improve doc consistency (#20184)
+* drivers: make use of busy_wait() (#20258)
+
+Documentation
+-------------
+
+* boards/nucleo-l433rc: doc update - image and MCU table (#20098)
+* boards/nucleo-l452re: doc update (#20207)
+* boards/p-nucleo-wb55: documentation update (#20115)
+* doc/mainpage: don't claim to be a microkernel (#20246)
+* doc: Extend and take responsibility for 802.15.4 security roadmap (#20123)
+* doc: Move roadmap in from the wiki (#20122)
+
+Build System / Tooling
+----------------------
+
++ dist/testbed-support: Add openmote board (#19979)
++ dist/tools/riotboot_gen_hdr/genhdr: add update command (#20239)
+* Bump version of UF2 package (#20035)
+* examples/gnrc_border_router: enable setting ZEP topology (#20006)
+* make all: document targets for RIOT base module (#20187)
+* make help: show generate-Makefile.ci (#20186)
+* tools/zep_dispatch: enhancements and fixes (#19996)
+
+Kconfig
+-------
+
+* tests/build_system: drop Kconfig tests (#20229)
+
+Examples
+--------
+
+* examples/dtls-sock: Enable non 32-bit architectures (#20196)
+* examples/dtls-sock: make use of helper functions (#20125)
+
+Testing
+-------
+
++ tests/periph/uart_locate_pins: new test/utility app (#20253)
++ tests/pkg/minmea: adding GGA, GSA and RMC sentences to test
+  application (#20005)
++ tests/unittests: add a unit test for ztimer (#20011)
+* .murdock: drop Kconfig compile check (#20211)
+* ci: ensure "xxx succeeded" properly fail (#20288)
+* ci: make native tests show up in "tests" (#20269)
+* tests/drivers/at: fix device table overflow (#20025)
+* tests/net/gnrc_sixlowpan_frag_sfr_congure_impl: disable test on
+  native (#20151)
+* tests/periph/uart: allow to set line ending (#20128)
+* tests/pkg/minmea: fixing RMC timestamp (#20018)
+* tests/sys/psa_crypto*: Fix failing tests (#20178)
+* tests/sys/psa_crypto_se_cipher: disable test on esp32-wroom-32 (#20152)
+* tests/sys/psa_crytpo_se: disable test on esp32-wroom-32 (#20150)
+* tests/unittests: blacklist LLVM on native (#19958)
+* tests: Slightly increase coverage of fmt unittests (#20272)
+
+API Changes
+-----------
+
++ print: Added size_t print format specifier (#20194)
+* drivers/mtd: use XFA for pointers to defined MTDs (#19465)
```

Comment:
That's a breaking change that requires the user to adapt their code - the PR contains a short description of what migration steps need to be performed.